src/Adapter/Order/CommandHandler/DeleteCartRuleFromOrderHandler.php

```diff
@@ -70,11 +69,6 @@ public function handle(DeleteCartRuleFromOrderCommand $command)
             throw new OrderException('Invalid cart provided.');
         }
 
-        $cartRule = new CartRule($orderCartRule->id_cart_rule);
-        if (!Validate::isLoadedObject($cartRule)) {
-            throw new OrderException('Invalid cart rule provided.');
-        }
-
```

Comment:
I removed this check, because the deletion of an OrderCartRule should not be depending on the fact that its related CartRule still exists.

Another solution would have been to implement a soft delete mechanism on the cart rule deletion mechanism (when we delete a CartRule in the BO). This solution seems less invasive.